### PR TITLE
Fix cloudwatch logs

### LINKED ISSUE #237: Update cloudwatch log group to match Airflow Role
### Are you sure this is not a new requirement or bug?

Yes

### Task Type

Sub-task

### 💡 Description

The registry-sweeper log group is not accessible anymore from Airflow. I am not sure how it used to work, possibly the role was updated manually and the specific manual update of the role was overiden by an automated deployment of nucleus which happened 06/05.

The log group needs to be renamed in the terraform scripts.

### PR DESCRIPTION
**Pull request by tloubrieu-jpl**
## 🗒️ Summary

Sweepers cloudwatch logs were not accessible from MWAA, possibly because a manual update the the Airflow role was lost after a terraform deployment of Nucleus.

The solution was to rename the cloudwatch log group.
Also to avoid discrepancies, between the log group and streams defined in the terraform and in the DAG's log configuration, the terraform out is now used to generate this part of the DAGs

### 🤖 AI Assistance Disclosure
- [ ] No AI assistance used
- [ ] AI used for light assistance (e.g., suggestions, refactoring, documentation help, minor edits)
- [ ] AI used for moderate content generation (AI generated some code or logic, but the developer authored or heavily revised the majority)
- [x] AI generated substantial portions of this code

Estimated % of code influenced by AI: _99_ %

## ⚙️ Test Data and/or Report

Tested with a deployment of sweeper with terraform and the DAG in Airflow in the AWS Dev account.

## ♻️ Related Issues

Fixes https://github.com/NASA-PDS/registry-sweepers/issues/237

## 🤓 Reviewer Checklist

*Reviewers: Please verify the following before approving this pull request.*

### Documentation and PR Content
- [ ] **Documentation:** README, Wiki, or inline documentation (Sphinx, Javadoc, Docstrings) have been updated to reflect these changes.
- [ ] **Issue Traceability:** The PR is linked to a valid GitHub Issue
- [ ] **PR Title:** The PR title is "user-friendly" clearly identifying what is being fixed or the new feature being added, that if you saw it in the Release Notes for a tool, you would be able to get the gist of what was done.

### Security & Quality
- [ ] **SonarCloud:** Confirmed no new High or Critical security findings.
- [ ] **Secrets Detection:** Verified that the Secrets Detection scan passed and no sensitive information (keys, tokens, PII) is exposed.
- [ ] **Code Quality:** Code follows organization style guidelines and best practices for the specific language (e.g., PEP 8, Google Java Style).

### Testing & Validation
- [ ] **Test Accuracy:** Verified that test data is accurate, representative of real-world PDS4 scenarios, and sufficient for the logic being tested.
- [ ] **Coverage:** Automated tests cover new logic and edge cases.
- [ ] **Local Verification:** (If applicable) Successfully built and ran the changes in a local or staging environment.

### Maintenance
- [ ] **Backward Compatibility:** Confirmed that these changes do not break existing downstream dependencies or API contracts (or that breaking changes are clearly documented).

### CODE REVIEW

File: terraform/generate_dags.py [hunk 1]
```diff
@@ -144,7 +152,8 @@ def main():
     output_dir.mkdir(parents=True, exist_ok=True)
 
     for node_name, task_arn in task_arns.items():
-        content = generate_dag(node_name, task_arn, env)
+        log_group = log_group_names[node_name]
```
Comment (nutjob4life) [inline terraform/generate_dags.py:155]: Should we handle the possible `KeyError` thrown if the `task_definition_arns` and `log_group_names` ever differ?
```python
try:
    log_group = log_group_names[node_name]
except KeyError:
    print(f"Missing log group name for node {node_name}", file=sys.stderr)
    sys.exit(1)
```

File: terraform/main.tf [hunk 2]
```diff
@@ -48,7 +48,7 @@ locals {
 resource "aws_cloudwatch_log_group" "registry_sweepers" {
   for_each = var.nodes
 
-  name              = "/ecs/pds-registry-sweepers-${each.key}-task"
+  name              = "/pds/ecs/pds-registry-sweepers-${each.key}-task"
```
Comment (nutjob4life) [inline terraform/main.tf:51]: This name change could leave behind log groups with the old name unless separately cleaned up.

**Review by nutjob4life — APPROVED** [review]
Code delta: ✓
Tests: not performed
Approval: ✅
Details: I did have a couple things to point out (`KeyError` and old log groups from name change) in the comments interspersed in the review, but I don't think they should block merging. See what you think and ignore them if you like!

File: terraform/main.tf (hunk 2 above)
Comment (tloubrieu-jpl) [reply terraform/main.tf:51]: Thanks @nutjob4life , I cleaned them in Dev. We should be good here.

File: terraform/generate_dags.py (hunk 1 above)
Comment (tloubrieu-jpl) [reply terraform/generate_dags.py:155]: For now I will keep it as-is. The list of nodes is an input of the terraform script which generates the log_groups and task definitions. The current script pulls the node list and task and log_group maps from the terraform, so there should not be discrepancies.
